cmd/meroxa/root/apps/deploy.go

```diff
@@ -524,6 +525,62 @@ func (d *Deploy) validate(ctx context.Context) error {
 	return turbineCLI.ValidateBranch(ctx, d.logger, d.path)
 }
 
+func (d *Deploy) checkResourceAvailability(ctx context.Context) error {
+	d.logger.Infof(ctx, "Checking resource availability for application %q (%s) before deployment...", d.appName, d.lang)
```

Comment:
we could probably leverage the use of `d.logger.StartSpinner` here to indicate it that way, what do you think? @jayjayjpg